### ISSUE TITLE
W0221 `arguments-differ`: why checking parameter naming?

### ISSUE THREAD

**Issue by pierrecdn** [PyCQA/pylint#3638]
The `arguments-differ` check is complaining when using a ABC + implicit Generics pattern, involving parameters renaming.
Also asking for an opinion here, this has probably already been discussed here, or in a PEP defined best practices, but I was not able to find where.

### Steps to reproduce
```
class Animal:
   def has_property():
     pass

class Bird(Animal):
   pass

class AbstractGrocery(ABC):
   @abstractmethod
   def deliver_food(self, animal)

class ConcreteBirdGrocery(AbstractGrocery):
    def deliver_food(self, bird):  # adapt the parameter naming for clarity sake
       pass
```
While untyped here and so subject to many confusions, adding type hints would give the same, valid in my opinion.

### Current behavior
`W0221 Parameters differ from overridden 'deliver_food' method [pylint]`

Most probably implemented here: https://github.com/PyCQA/pylint/blob/1fc490c0a1d0df5d4d6a64da0ccf2dd711b9bf67/pylint/checkers/classes.py#L279-L280

### Expected behavior

Since Python doesn't support function overloading, I'd say that as long as the number of args is respected, that's OK.
The naming choice could be bad, but that's another story and out of the scope of pylint anyway.

### pylint --version output
2.4.4

**Comment by PCManticore** [comment]
That is intended (but ideally would be a separate check).

Imagine you have the following class:
```
class A:
    def method(self, some_arg):
          pass

class B(A):
   def method(self, some_other_arg):
            pass
```

If `A.method()` would be called with keyword arguments, as in `method(some_arg=...)`, changing the signature in `B.method` means that now `B` cannot fully represent the base class in the call sites where the method is called with keyword arguments, leading to potential bugs or issues that need to be addressed with refactoring. It also violates Liskov substitution principle in case the hierarchy of objects that implement this method is bigger.

Hope this makes sense. As mentioned, ideally it should be a separate check, but that's the case now.

**Comment by pierrecdn** [comment]
> If A.method() would be called with keyword arguments, as in method(some_arg=...), changing the signature in B.method means that now B cannot fully represent the base class in the call sites where the method is called with keyword arguments, leading to potential bugs or issues that need to be addressed with refactoring.

That's a very fair point indeed!

> ideally it should be a separate check

True, if ever I decide to disable this check because I consider this is acceptable in my use-case, it would make the change more targeted to variable renaming, nothing else.

Thanks!
